api/editUser.go

```diff
@@ -0,0 +1,46 @@
+package editUser
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/mhghw/fara-message/db"
+)
+
+func editUser(c *gin.Context) {
+	var username string
+
+	err := c.BindJSON(&username)
+	if err != nil {
+		fmt.Errorf("error binding JSON:%w", err)
+		c.Status(400)
+		return
+	}
+
+	if len(username) < 3 {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": "the username or password is incorrect",
+		})
+		return
+	}
+
+	userUnderReview, err1 := db.UsersDB.GetUserByUsername(username)
+	if err1 != nil {
```

Comment:
You didn't fix the err1
name it as err
and also you didn't use the err value itself, maybe the database connection is not available and you got an error, we can't find the err with this approach